templates/boltzmann/middleware/honeycomb.ts

```diff
@@ -9,77 +19,179 @@ import beeline from 'honeycomb-beeline'
 import isDev from 'are-we-dev'
 void `{% endif %}`;
 
+let otelTracer: OtelTracer | null = null;
+
+const OTEL_NAMESPACE = 'boltzmann'
+const OTEL_REQ_NAMESPACE = `${OTEL_NAMESPACE}.request`
+const OTEL_REQ_QUERY = `${OTEL_REQ_NAMESPACE}.query`
+const OTEL_REQ_PARAM_NAMESPACE = `${OTEL_REQ_NAMESPACE}.param`
+
+function getOtelTracer() {
+  if (!otelTracer) {
+    otelTracer = otelTrace.getTracer('boltzmann', '1.0.0')
+  }
+  return otelTracer
+}
+
 function trace ({
   headerSources = ['x-honeycomb-trace', 'x-request-id'],
 } = {}) {
-  if (!process.env.HONEYCOMB_WRITEKEY) {
+  if (!isHoneycomb(process.env)) {
     return (next: Handler) => (context: Context) => next(context)
   }
 
-  const schema = require('honeycomb-beeline/lib/schema')
-  const tracker = require('honeycomb-beeline/lib/async_tracker')
-
-  return function honeycombTrace (next: Handler) {
-    return (context: Context) => {
-      const traceContext = _getTraceContext(context)
-      const trace = beeline.startTrace({
-        [schema.EVENT_TYPE]: 'boltzmann',
-        [schema.PACKAGE_VERSION]: '1.0.0',
-        [schema.TRACE_SPAN_NAME]: `${context.method} ${context.url.pathname}${context.url.search}`,
-        [schema.TRACE_ID_SOURCE]: traceContext.source,
-        'request.host': context.host,
-        'request.original_url': context.url.href,
-        'request.remote_addr': context.remote,
-        'request.method': context.method,
-        'request.scheme': context.url.protocol,
-        'request.path': context.url.pathname,
-        'request.query': context.url.search
-      },
-      traceContext.traceId,
-      traceContext.parentSpanId,
-      traceContext.dataset)
-
-      if (isDev()) {
-        context._honeycombTrace = trace
-      }
-
-      if (traceContext.customContext) {
-        beeline.addContext(traceContext.customContext)
-      }
-
-      if (!trace) {
-        return next(context)
-      }
-
-      const boundFinisher = beeline.bindFunctionToTrace((response: ServerResponse) => {
-        beeline.addContext({
-          'response.status_code': String(response.statusCode)
+  if (isOtel(process.env)) {
+    return function honeycombOtelTrace (next: Handler) {
+      return async (context: Context) => {
+        const tracer = getOtelTracer()
+        let carrier = {}
+
+        // Start a parent span
+        const parentSpan = tracer.startSpan(`${context.method} ${context.url.pathname}${context.url.search}`, {
```

Comment:
we need to pass in the `parentContext` here (currently down in line 76) in order for trace propagation to work properly.